core/IssueQueue.cpp

```diff
@@ -0,0 +1,228 @@
+// <IssueQueue.cpp> -*- C++ -*-
+
+#include "IssueQueue.hpp"
+#include "CoreUtils.hpp"
+namespace olympia {
+const char IssueQueue::name[] = "issue_queue";
+
+IssueQueue::IssueQueue(sparta::TreeNode *node, const IssueQueueParameterSet *p)
+    : sparta::Unit(node),
+      reg_file_(olympia::coreutils::determineRegisterFile(
+          node->getGroup().substr(node->getGroup().find("_") + 1))),
+      scheduler_size_(p->scheduler_size), in_order_issue_(p->in_order_issue) {
+  in_execute_inst_.registerConsumerHandler(CREATE_SPARTA_HANDLER_WITH_DATA(
+      IssueQueue, getInstsFromDispatch_, InstPtr));
+  in_exe_pipe_done_.registerConsumerHandler(
+      CREATE_SPARTA_HANDLER_WITH_DATA(IssueQueue, readyExeUnit_, uint32_t));
+  in_reorder_flush_.registerConsumerHandler(CREATE_SPARTA_HANDLER_WITH_DATA(
+      IssueQueue, flushInst_, FlushManager::FlushingCriteria));
+  sparta::StartupEvent(node,
+                       CREATE_SPARTA_HANDLER(IssueQueue, setupIssueQueue_));
+  node->getParent()
+      ->registerForNotification<bool, IssueQueue, &IssueQueue::onROBTerminate_>(
+          this, "rob_stopped_notif_channel",
+          false /* ROB maybe not be constructed yet */);
+}
+
+void IssueQueue::setupIssueQueue_() {
+  std::vector<core_types::RegFile> reg_files = {core_types::RF_INTEGER,
+                                                core_types::RF_FLOAT};
```

Comment:
_Technically_ in this new paradigm, an IssueQueue will be handed the register file types it supports -- most likely via parameter.

Reply:
Implemented in [6b0e363](https://github.com/riscv-software-src/riscv-perf-model/pull/140/commits/6b0e3633c1b7331170afda8920f596f9fd16b233) to use the reg_file_ parameter, which is defined by searching the node name.